core/beneficiary/saver.go

```diff
@@ -61,23 +63,77 @@ type l1Saver struct {
 	set     settler
 	ad      addressProvider
 	chainID int64
+	beneficiaryChangeKeeper
 }
 
-func newL1Saver(chainID int64, ad addressProvider, bc multiChainBC, set settler) *l1Saver {
+func newL1Saver(chainID int64, ad addressProvider, set settler, st storage) *l1Saver {
 	return &l1Saver{
-		chainID: chainID,
-		set:     set,
-		ad:      ad,
+		chainID:                 chainID,
+		set:                     set,
+		ad:                      ad,
+		beneficiaryChangeKeeper: beneficiaryChangeKeeper{st: st, chainID: chainID},
 	}
 }
 
+// SettleWithBeneficiaryState represents the state of settle with beneficiary transaction
+type SettleWithBeneficiaryState string
```

Comment:
gonna nag a little: the word `beneficiary` is already in the package name, so when you call this, you'll now call it like `beneficiary.SettleWithBeneficiaryState` I guess something like `beneficiary.SettlementState` would make more sense?